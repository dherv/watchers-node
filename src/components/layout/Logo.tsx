import React from "react";
import styled from "styled-components";

const Logo = () => <Heading>watchers</Heading>;

const Heading = styled.h1`
  font-size: 2rem;
  font-weight: 600;
  text-transform: capitalize;
`;

export default Logo;
