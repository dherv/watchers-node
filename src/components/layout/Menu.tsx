import React from "react";
import ActiveLink from "./ActiveLink";
import styled from "styled-components";

const Menu = () => (
  <List>
    <ListItem>
      <ActiveLink href="/movies">movies</ActiveLink>
    </ListItem>
    <ListItem>
      <ActiveLink href="/series">series</ActiveLink>
    </ListItem>
    <ListItem>
      <ActiveLink href="/watchlist">watchlist</ActiveLink>
    </ListItem>
    <ListItem>
      <ActiveLink href="/likelist">likelist</ActiveLink>
    </ListItem>
  </List>
);

const List = styled.ul`
  display: flex;
  margin-left: 8rem;
`;

const ListItem = styled.li`
  font-weight: 100;
  font-size: 1.1rem;
  margin: 0 4rem;
`;

export { List, ListItem };
export default Menu;
