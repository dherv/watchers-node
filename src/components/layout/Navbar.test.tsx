import { shallow } from "enzyme";
import React from "react";
import Navbar, { Nav } from "./Navbar";
import Menu from "./Menu";
import Logo from "./Logo";

describe("Navbar component", () => {
  const wrapper = shallow(<Navbar />);
  describe("UI", () => {
    test("should display one Nav component", () => {
      expect.assertions(1);
      expect(wrapper.find(Nav)).toHaveLength(1);
    });
    test("should display one Menu component", () => {
      expect.assertions(1);
      expect(wrapper.find(Menu)).toHaveLength(1);
    });
    test("should display one Logo component", () => {
      expect.assertions(1);
      expect(wrapper.find(Logo)).toHaveLength(1);
    });
  });
});
