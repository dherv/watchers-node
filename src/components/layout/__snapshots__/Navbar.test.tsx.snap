// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component snapshot should match 1`] = `
.c2 {
  height: 60px;
  line-height: 60px;
  width: 100%;
}

.c1 {
  font-weight: 100;
  font-size: 1.1rem;
  margin: 0 4rem;
  position: relative;
}

.c1::after {
  content: "";
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  border-width: 2px;
  border-style: solid;
  border-color: transparent;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  margin-left: 7rem;
}

<ul
  className="c0"
>
  <li
    className="c1"
    href="/movies"
  >
    <a
      className="c2"
      height={60}
      href="/movies"
      onClick={[Function]}
    >
      movies
    </a>
  </li>
  <li
    className="c1"
    href="/series"
  >
    <a
      className="c2"
      height={60}
      href="/series"
      onClick={[Function]}
    >
      series
    </a>
  </li>
  <li
    className="c1"
    href="/watchlist"
  >
    <a
      className="c2"
      height={60}
      href="/watchlist"
      onClick={[Function]}
    >
      watchlist
    </a>
  </li>
  <li
    className="c1"
    href="/likelist"
  >
    <a
      className="c2"
      height={60}
      href="/likelist"
      onClick={[Function]}
    >
      likelist
    </a>
  </li>
</ul>
`;
